### DISCUSSION TITLE
Create base folder in zip file with FastZip

### DISCUSSION THREAD

**Discussion by frankzoldan** [icsharpcode/SharpZipLib#694] [Q&A]
I do like **FastZip** because I can create zip files and store a source directory recursively with little coding.
However, I could not find out how to create a base directory in the zip file where I would store all the files and folders of the source directory.

When I have some files and folders like this:
>*d:\Projects*\\**Project1\Sub1\File1.txt**
>*d:\Projects*\\**Project1\Sub2\File1.txt**
>*d:\Projects*\\**Project1\Sub1\File2.txt**
>*d:\Projects*\\**Project1\Sub2\File2.txt**

And I create a zip file with:
`fastZip.CreateZip(@"d:\Backups\BackupA.zip", @"d:\Projects\Project1\", true, "", "");`

Then the destination paths will be:
>*d:\Backups\BackupA.zip*\\**Sub1\File1.txt**
>*d:\Backups\BackupA.zip*\\**Sub2\File1.txt**
>*d:\Backups\BackupA.zip*\\**Sub1\File2.txt**
>*d:\Backups\BackupA.zip*\\**Sub2\File2.txt**

But I want them to be:
>*d:\Backups\BackupA.zip*\\**Project1\Sub1\File1.txt**
>*d:\Backups\BackupA.zip*\\**Project1\Sub2\File1.txt**
>*d:\Backups\BackupA.zip*\\**Project1\Sub1\File2.txt**
>*d:\Backups\BackupA.zip*\\**Project1\Sub2\File2.txt**

**Project1** should be the base folder for the archived files and folders.

Is this possible to achieve with FastZip?

**Comment by ZephyrDogma** [comment]
I too have this problem and would like to create a base folder within my zip

**Comment by piksel** [comment]
You can use the parent directory as the base and then use a path filters to only include the `Project1` directory:

```cs
fastZip.CreateZip(@"d:\Backups\BackupA.zip", @"d:\Projects\", true, @"Project1\\.*", @"Project1\\.*");
```
